pkg/auth/dependency/hook/provider.go

```diff
@@ -1,12 +1,15 @@
 package hook
 
 import (
+	"net/url"
+
 	"github.com/skygeario/skygear-server/pkg/auth/event"
 	"github.com/skygeario/skygear-server/pkg/auth/model"
 	"github.com/skygeario/skygear-server/pkg/core/handler"
 )
 
 type Provider interface {
+	SetBaseURL(*url.URL)
```

Comment:
It seems that it is possible to avoid this method. In `pkg/core/inject/inject.go`, `RequestDependencyMap.Provide` has access to `*http.Request` but `DependencyMap.Provide` hides it. Maybe we can remove such hiding.